x/auctions/keeper/msg_server_test.go

```diff
@@ -0,0 +1,387 @@
+				assert.NoError(t, err, "failed to update auctions params")
+			},
+			input: func() *types.MsgDepositCoin {
+				return &types.MsgDepositCoin{
+					Sender: validSenderKey.AccAddr.String(),
+					Amount: sdk.NewCoin(utils.BaseDenom, depositAmount),
+				}
+			},
+			postCheck:   func() {},
+			expErr:      true,
+			errContains: types.ErrAuctionDisabled.Error(),
+		},
+		{
+			name: "fail when sender is not valid bech32",
+			malleate: func() {
+			},
+			input: func() *types.MsgDepositCoin {
+				return &types.MsgDepositCoin{
+					Sender: "",
+					Amount: sdk.NewCoin(utils.BaseDenom, depositAmount),
+				}
+			},
+			postCheck:   func() {},
+			expErr:      true,
+			errContains: "invalid sender address",
+		},
+		{
+			name: "fail when sender does not have enough funds",
+			malleate: func() {
+			},
+			input: func() *types.MsgDepositCoin {
+				return &types.MsgDepositCoin{
+					Sender: emptyAddress.String(),
+					Amount: sdk.NewCoin(utils.BaseDenom, depositAmount),
+				}
+			},
+			postCheck:   func() {},
+			expErr:      true,
+			errContains: "transfer of deposit failed",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			keyring := testkeyring.New(1)
+			network = testnetwork.NewUnitTestNetwork(
+				testnetwork.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+			)
+
+			validSenderKey = keyring.GetKey(0)
+
+			// Update chain environment before executing the message.
+			tc.malleate()
+			_, err := network.App.AuctionsKeeper.DepositCoin(network.GetContext(), tc.input())
+
+			if tc.expErr {
+				assert.Error(t, err, "expected error but got nil")
+				assert.Contains(t, err.Error(), tc.errContains, "expected different error for test case: %s", tc.name)
+			} else {
+				assert.NoError(t, err, "error not expected")
+			}
+
+			tc.postCheck()
+		})
+	}
+}
+
+func TestUpdateParams(t *testing.T) {
+	var network *testnetwork.UnitTestNetwork
+
+	var eoaKey testkeyring.Key
+	authorityAddress := authtypes.NewModuleAddress(govtypes.ModuleName).String()
+
+	testCases := []struct {
+		name        string
+		preCheck    func()                        // used to modify the initial conditions of the test or for pre test checks
+		input       func() *types.MsgUpdateParams // return the message used for the bid
+		postCheck   func()
+		expErr      bool
+		errContains string
+	}{
+		{
+			name: "pass",
+			preCheck: func() {
+				params := network.App.AuctionsKeeper.GetParams(network.GetContext())
+				assert.Equal(t, params.EnableAuction, true)
+			},
+			input: func() *types.MsgUpdateParams {
+				params := network.App.AuctionsKeeper.GetParams(network.GetContext())
+				params.EnableAuction = false
+				return &types.MsgUpdateParams{
+					Authority: authorityAddress,
+					Params:    params,
+				}
+			},
+			postCheck: func() {
+				params := network.App.AuctionsKeeper.GetParams(network.GetContext())
+				assert.Equal(t, params.EnableAuction, false, "expected params to be updated")
+			},
```

Comment:
ditto as above - the post check function can be removed and just add a `expAuctionEnabled` to check against


```suggestion
			expAuctionEnabled: false
```